vine/src/tools/repl.rs

```diff
@@ -170,23 +171,21 @@ impl<'ctx, 'ivm, 'ext, 'comp> Repl<'ctx, 'ivm, 'ext, 'comp> {
       fn chart(&mut self, charter: &mut Charter<'_>) {
         let mut extractor = ExtractItems::default();
         extractor.visit(&mut *self.block);
-        for item in extractor.items {
-          charter.chart_item(VisId::Def(self.repl_mod), item, self.repl_mod, GenericsId::NONE);
-          self.extracted_items = true;
-        }
-      }
-
-      fn pre_resolve(&mut self, resolver: &mut Resolver<'_>) {
-        if self.extracted_items {
-          resolver.finder_cache.revert(&Checkpoint::default());
+        if !extractor.items.is_empty() {
```

Comment:
Is this `if` necessary? Shouldn't the for loop just not iterate on an empty collection?